hazelcast-oss/Dockerfile

```diff
@@ -36,23 +36,25 @@ COPY *.sh *.yaml *.jar *.properties ${HZ_HOME}/
 
 # Install
 RUN echo "Installing new APK packages" \
-    && apk add --no-cache openjdk11-jre bash curl \
+    && apk add --no-cache openjdk11-jre bash curl perl-xml-xpath \
     && echo "Downloading Hazelcast and related JARs" \
     && mkdir "${HZ_HOME}/lib" \
     && cd "${HZ_HOME}/lib" \
+    && HAZELCAST_ALL_URL=$(${HZ_HOME}/get-hz-all-url.sh) \
```

Comment:
Would be better to define an environment variable by using a dedicated keyword:

https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#env

Reply:
Also, you should not need to `$(${HZ_HOME}/get-hz-all-url.sh)`, it should be enough to just:

> HAZELCAST_ALL_URL="${HZ_HOME}/get-hz-all-url.sh"

Reply:
I actually tried running `ENV HAZELCAST_ALL_URL=$(${HZ_HOME}/get-hz-all-url.sh)`. However the ENV variable's value turned out to be `$(/opt/hazelcast/get-hz-all-url.sh)`. So I don't think we can run commands in the `ENV` step.

Even if we could run commands in `ENV` step, there is also one downside to using `ENV` as a seperate step because that would split the installation `RUN` step into two parts. I am not sure how much bigger the image size would get but I believe it would be significant.

Reply:
Ah, wait. You're running the script and not defining the environment variable. Ignore my comment :)